src/Applications/LDAS_App/preprocess_ldas_routines.F90

```diff
@@ -2438,7 +2438,7 @@ subroutine optimize_latlon(fname_tilefile, N_proc_string)
        
        if( sum(local_land) /= total_land) stop ("wrong distribution")
        if( sum(IMS) /= N_lon) stop ("wrong domain distribution")
-       
+       if( any(IMS <=1) ) stop ("Ask for too many processors. Each processor should have at least 2 grid cells")  
```

Comment:
@weiyuan-jiang, 
Thanks for taking a crack at this issue.  This fix isn't quite what I had thought was needed.  
Let's say we end up with: 
`IMS = [ 2 2 2 2 1 3 2 2 ]`
for 16 longitude strips on 8 processors, then a simple solution would be to re-assign one longitude strip so we have 
`IMS = [ 2 2 2 2 2 2 2 2 ]`
This is perhaps slightly less optimal than if we were able to allow `IMS(5)=1`, but it should be serviceable.  As implemented now, the user's only option would be to use, say, 4 processors and probably end up with something like:
`IMS = [ 4 4 5 3]`
I think for JMS on the cube-sphere tile space, we already have a fix for JMS(k)=0 or JMS(k)=1 (~line 2235):
```
if(JMS(i) == 0) then
   n = maxloc(JMS(n1:n2),DIM=1)
   JMS(i) = 1
   JMS(n+n1-1) = JMS(n+n1-1)-1
endif
if(JMS(i) == 1) then
   n = maxloc(JMS(n1:n2),DIM=1)
   JMS(i) = 2
   JMS(n+n1-1) = JMS(n+n1-1)-1
endif
```